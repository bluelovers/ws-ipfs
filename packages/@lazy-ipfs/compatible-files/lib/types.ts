import { IPFS } from 'ipfs-core-types';
import { IFilesStatOptionsExtra } from './stat';

export type IOptionsExtra<O extends {}> = O & {
	ignoreError?: boolean,
	returnStat?: boolean,
	statOptions?: IFilesStatOptionsExtra,
}

export type IExtendOptions<T extends {}, O extends {} = {}, TO extends {} = {}> = T & {
	extraOptions?: IOptionsExtra<O>
}

export type IHasFilesAPI = Pick<IPFS, 'files'>
