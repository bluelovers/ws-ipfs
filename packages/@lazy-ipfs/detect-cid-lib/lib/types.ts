import JsCID from 'cids';
import { CID as MultiformatsCID } from 'multiformats';
import { IRawJsCID } from './js-cids';
import { IRawMultiformatsCID } from './js-multiformats';

export type ICIDObject = JsCID | MultiformatsCID;
export type ICIDRawObject = IRawJsCID | IRawMultiformatsCID;

export type IRawCIDVersion = ICIDObject["version"];
export type ICIDValue = ICIDObject | string;
export type ICIDValueOrRaw = ICIDValue | ICIDRawObject;

export type I_CIDLike = Pick<ICIDObject, 'version' | 'multihash'>
