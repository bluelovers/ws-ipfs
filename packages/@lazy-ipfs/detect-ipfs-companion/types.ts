import { IPFS } from 'ipfs-core-types';

declare global
{
	interface Window
	{
		ipfsCompanion?: {
			ipfs?: IPFS,
		}
		ipfs?: IPFS,
		chrome: Window["chrome"] & {

		}
	}
}
export type IWindow = Window & typeof globalThis;

export type IDetectIpfsCompanionSyncParams = {
	window?: IWindow
};

export type IDetectIpfsWindowParams = {
	window?: {
		ipfs?: IPFS,
	}
};
