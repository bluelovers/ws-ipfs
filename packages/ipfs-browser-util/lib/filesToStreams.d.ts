import { IFile } from 'pull-file-reader2/types';
export interface IFilesToStreams {
    path: string;
    content(end: any, cb: any): any;
    size: number;
}
export declare function filesToStreams(files: IFile[]): Promise<IFilesToStreams[]>;
export default filesToStreams;
