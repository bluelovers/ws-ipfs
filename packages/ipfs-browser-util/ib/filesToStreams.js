"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.filesToStreams = void 0;
const pull_file_reader2_1 = __importDefault(require("pull-file-reader2"));
async function filesToStreams(files) {
    const streams = [];
    for (const file of files) {
        const stream = pull_file_reader2_1.default(file);
        streams.push({
            // @ts-ignore
            path: file.filepath || file.webkitRelativePath || file.name,
            content: stream,
            size: file.size,
        });
    }
    return streams;
}
exports.filesToStreams = filesToStreams;
exports.default = filesToStreams;
//# sourceMappingURL=filesToStreams.js.map