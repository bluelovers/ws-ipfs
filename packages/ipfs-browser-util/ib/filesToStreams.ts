import fileReader from 'pull-file-reader2'

export async function filesToStreams(files: File[])
{
	const streams: {
		path: string,
		content,
		size: number,
	}[] = []

	for (const file of files)
	{
		const stream = fileReader(file)

		streams.push({
			// @ts-ignore
			path: file.filepath || file.webkitRelativePath || file.name,
			content: stream,
			size: file.size,
		})
	}

	return streams
}

export default filesToStreams
