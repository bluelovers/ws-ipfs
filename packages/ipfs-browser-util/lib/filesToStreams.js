"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.filesToStreams = void 0;
const tslib_1 = require("tslib");
const pull_file_reader2_1 = (0, tslib_1.__importDefault)(require("pull-file-reader2"));
/**
 * https://github.com/ipfs-shipyard/ipfs-webui/blob/master/src/lib/files.js
 */
async function filesToStreams(files) {
    const streams = [];
    for (const file of files) {
        const stream = (0, pull_file_reader2_1.default)(file);
        streams.push({
            path: file.filepath || file.webkitRelativePath || file.name,
            content: stream,
            size: file.size,
        });
    }
    return streams;
}
exports.filesToStreams = filesToStreams;
exports.default = filesToStreams;
//# sourceMappingURL=filesToStreams.js.map