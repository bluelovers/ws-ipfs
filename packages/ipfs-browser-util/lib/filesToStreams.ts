import fileReader from 'pull-file-reader2'
import { IFile, IFileLike } from 'pull-file-reader2/types';
import { isSource } from 'pull-file-reader2/isSource';

export interface IFilesToStreams
{
	path: string,
	content(end, cb),
	size: number,
}

export async function filesToStreams(files: IFile[])
{
	const streams: IFilesToStreams[] = []

	for (const file of files as IFileLike[])
	{
		const stream = fileReader(file)

		streams.push({
			path: file.filepath || file.webkitRelativePath || file.name,
			content: stream,
			size: file.size,
		})
	}

	return streams
}

export default filesToStreams
