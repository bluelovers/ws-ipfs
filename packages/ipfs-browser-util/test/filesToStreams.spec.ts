/**
 * Created by user on 2020/3/29.
 */

import { filesToStreams } from '..';
import { IFile } from 'pull-file-reader2/types';
import { IFilesToStreams } from '../lib/filesToStreams';

it('drop one directory', async () =>
{
	const input: IFile[] = [
		{
			size: 215786,
			filepath: 'Dir/T.pdf',
			webkitRelativePath: 'Dir/T.pdf',
		},
		{
			size: 1511696,
			filepath: 'Dir/T2.pdf',
			webkitRelativePath: 'Dir/T2.pdf',
		},
	]

	const expected: Partial<IFilesToStreams>[] = [
		{
			size: 215786,
			path: 'Dir/T.pdf',
		},
		{
			size: 1511696,
			path: 'Dir/T2.pdf',
		},
	]

	const output = await filesToStreams(input)

	expect(output).toMatchObject(expected)
	expect(output).toMatchSnapshot()
});

