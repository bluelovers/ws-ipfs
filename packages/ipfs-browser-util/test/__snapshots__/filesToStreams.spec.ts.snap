// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drop one directory 1`] = `
Array [
  Object {
    "content": [Function],
    "path": "Dir/T.pdf",
    "size": 215786,
  },
  Object {
    "content": [Function],
    "path": "Dir/T2.pdf",
    "size": 1511696,
  },
]
`;
